### DISCUSSION TITLE
Transaction time

### DISCUSSION THREAD

**Discussion by Snyaify** [firefly-iii/firefly-iii#5318] [Firefly III] [answered]
Hello, 
I ran into a problem when editing a created transaction that has a date and time of creation. As soon as I make any changes, the transaction time is reset by default to 00:00:00.
Is it possible to set the transaction time by myself? (Otherwise, the order is violated) Or, when editing a transaction, add the option to not reset time? (time of transaction is imported from csv bank file)

**Comment by JC5** [comment, marked as answer]
This will be possible in v2!

**Reply by MacPaille** [reply]
Quick question related to time, will it be possible to have this field available in the csv-importer?

**Reply by JC5** [reply]
Is already the case, just set a date stamp that includes the time.

**Reply by MacPaille** [reply]
Indeed I forgot that field has to be defined using the date PHP function. Thanks!

**Reply by Still34** [reply]
I'm guessing it's still not available in the current release yet?

**Reply by diamant-x** [reply]
I confirm that currently it is possible using FiDi to set the time too. I use FiDi from CSV with the correct date time setting as seen in the 'preview' screen of FiDi (following options linked in the online docs: https://www.php.net/manual/en/datetime.format.php) :
![image](https://user-images.githubusercontent.com/7346338/164429854-3f32aa71-e109-48d0-ab28-9fad7ea75c9b.png)
And when it is imported, the time is correctly set:
![image](https://user-images.githubusercontent.com/7346338/164430352-b315e011-ffeb-4033-a788-184f930bd100.png)

But I have the same issue as the original poster, as soon as any manual change is done in the transaction (even if the date field is not changed, only a category) the time detail is lost and reset to 00:00:00 of the same day. Looking forward for v2 fix of this.
